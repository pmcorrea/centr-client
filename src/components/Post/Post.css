
.Post {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  height: 88px;
  background-color: #eee;
  color: #cacaca;
  width: 90vw;
  padding: 15px;
  margin: 10px auto;
  border-radius: 5px;
  box-shadow: 0px 5px 20px rgba(128, 128, 128, 0.733);
}

.Post__edit_button {
  background-color: transparent;
  border: 0;
  color: black;
  float: right;
  font-size: 12px;
  padding:0px;
}

.Post__edit_button:hover {
  color: orange;
}

.Post__delete_button {
  background-color: transparent;
  border: 0;
  color: black;
  font-size: 20px;
  height: 20px;
  padding:0px;
}

.Post__delete_button:hover {
  color: orange;
}

.Post__edit_button a {
  color: black;
  text-decoration: none;
}

.Post__title {
  margin: 0px 0px 10px;
  font-size: 15px;
}

.Post__title a {
  color: rgb(112, 39, 161);
  display: block;
  text-decoration: none;
}
.Post__title a:hover {
  color: orange;
}

.Post__info {
  color: black;
  font-size: 14px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: left;
  height: auto;
}

.Post__visibility{
  font-weight: bold;
  margin: 0px;
}

.Post__buttons_container{
  width: 80px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  height: 30px;
}
