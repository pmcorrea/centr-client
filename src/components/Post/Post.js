import React, { Component } from "react";
import { Link } from "react-router-dom";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
// Context and service
import MainContext from "../../contexts/MainContext";
import AuthApiService from "../../services/auth-api-service.js"
// Style
import "./Post.css";

export default class Post extends Component {
	static contextType = MainContext;

	handleDeletePost(value) {
		AuthApiService.handleDeletePost(value)
		.then(result => {
			this.context.getDataWithToken()
		})
		.catch(error => {
			console.error(error)
		})  
	}

	render() {
		let modified = new Date(this.props.modified)
		let date = modified.getDate()
		let month = modified.getMonth()
		let year = modified.getFullYear()

		return (
			<>
			<div className="Post">
				<div className="Post__info">
					<h2 className="Post__title">
						<Link to={`/post/${this.props.id}`}>{this.props.name}</Link>
					</h2>

					<div className="Post__dates-modified">
						<span className="Date">
							{`${month}/${date}/${year}`}
						</span>
					</div>

					<div className="Post__visibility_div">
						<p className="Post__visibility">
							{this.props.visibility}
						</p>
					</div>
				</div>

				<div className="Post__buttons_container">
					<button
						className="Post__edit_button"
						type="button">
						<Link to={`/edit/${this.props.id}`}>
							<FontAwesomeIcon className="Folder__delete" icon="edit"/>
						</Link>
					</button>

					<button
						className="Post__delete_button"
						type="button"
						onClick={() => this.handleDeletePost(this.props.id)}>
						<FontAwesomeIcon className="Folder__delete" icon="trash-alt"/>
					</button>
				</div>
			</div>
			</>
		);
	}
}
