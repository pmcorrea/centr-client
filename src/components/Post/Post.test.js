import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Post from './Post'

describe(`Post component`, () => {
  const props = {
    id: 'a',
    name: 'test-class-name',
    modified: new Date(2018, 12, 15),
  }

  it('renders a .Post by default', () => {
    const wrapper = shallow(<Post />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the Post given props', () => {
    const wrapper = shallow(<Post {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
