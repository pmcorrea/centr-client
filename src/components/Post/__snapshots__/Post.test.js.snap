// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component renders a .Post by default 1`] = `
<Fragment>
  <div
    className="Post"
  >
    <div
      className="Post__info"
    >
      <h2
        className="Post__title"
      >
        <Link
          replace={false}
          to="/post/undefined"
        />
      </h2>
      <div
        className="Post__dates-modified"
      >
        <span
          className="Date"
        >
          NaN/NaN/NaN
        </span>
      </div>
      <div
        className="Post__visibility_div"
      >
        <p
          className="Post__visibility"
        />
      </div>
    </div>
    <div
      className="Post__buttons_container"
    >
      <button
        className="Post__edit_button"
        type="button"
      >
        <Link
          replace={false}
          to="/edit/undefined"
        >
          <FontAwesomeIcon
            border={false}
            className="Folder__delete"
            fixedWidth={false}
            flip={null}
            icon="edit"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
        </Link>
      </button>
      <button
        className="Post__delete_button"
        onClick={[Function]}
        type="button"
      >
        <FontAwesomeIcon
          border={false}
          className="Folder__delete"
          fixedWidth={false}
          flip={null}
          icon="trash-alt"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </button>
    </div>
  </div>
</Fragment>
`;

exports[`Post component renders the Post given props 1`] = `
<Fragment>
  <div
    className="Post"
  >
    <div
      className="Post__info"
    >
      <h2
        className="Post__title"
      >
        <Link
          replace={false}
          to="/post/a"
        >
          test-class-name
        </Link>
      </h2>
      <div
        className="Post__dates-modified"
      >
        <span
          className="Date"
        >
          0/15/2019
        </span>
      </div>
      <div
        className="Post__visibility_div"
      >
        <p
          className="Post__visibility"
        />
      </div>
    </div>
    <div
      className="Post__buttons_container"
    >
      <button
        className="Post__edit_button"
        type="button"
      >
        <Link
          replace={false}
          to="/edit/a"
        >
          <FontAwesomeIcon
            border={false}
            className="Folder__delete"
            fixedWidth={false}
            flip={null}
            icon="edit"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
        </Link>
      </button>
      <button
        className="Post__delete_button"
        onClick={[Function]}
        type="button"
      >
        <FontAwesomeIcon
          border={false}
          className="Folder__delete"
          fixedWidth={false}
          flip={null}
          icon="trash-alt"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </button>
    </div>
  </div>
</Fragment>
`;
